### PR TITLE
Save edited HTML on configuration changed

### PR DESCRIPTION
**Pull request by LunarX**
When the html is edited by the user, we need to keep the changes and reload the edited html across config changes

Depends on #14

### CODE REVIEW

File: rich-html-editor/src/main/java/com/infomaniak/lib/richhtmleditor/EditorReloader.kt
Comment (sirambd) [inline rich-html-editor/src/main/java/com/infomaniak/lib/richhtmleditor/EditorReloader.kt:1]: Missing kDoc here

File: rich-html-editor/src/main/java/com/infomaniak/lib/richhtmleditor/EditorReloader.kt [hunk 1]
```diff
@@ -0,0 +1,40 @@
+package com.infomaniak.lib.richhtmleditor
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.launch
+
+class EditorReloader(private val viewModelScope: CoroutineScope) {
```
Comment (sirambd) [inline rich-html-editor/src/main/java/com/infomaniak/lib/richhtmleditor/EditorReloader.kt:7]:
```suggestion
class EditorReloader(private val coroutineScope: CoroutineScope) {
```

File: rich-html-editor/src/main/java/com/infomaniak/lib/richhtmleditor/EditorReloader.kt [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+    private var needToReloadHtml: Boolean = false
+    private var savedHtml = MutableStateFlow<String?>(null)
+
+    suspend fun load(editor: RichHtmlEditorWebView, defaultHtml: String) {
+        if (needToReloadHtml) {
+            savedHtml.collect {
```
Comment (sirambd) [inline rich-html-editor/src/main/java/com/infomaniak/lib/richhtmleditor/EditorReloader.kt:14]: `collectLatest` is better

File: rich-html-editor/src/main/java/com/infomaniak/lib/richhtmleditor/EditorReloader.kt [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.launch
+
+class EditorReloader(private val viewModelScope: CoroutineScope) {
+
+    private var needToReloadHtml: Boolean = false
+    private var savedHtml = MutableStateFlow<String?>(null)
+
+    suspend fun load(editor: RichHtmlEditorWebView, defaultHtml: String) {
+        if (needToReloadHtml) {
```
Comment (sirambd) [inline rich-html-editor/src/main/java/com/infomaniak/lib/richhtmleditor/EditorReloader.kt:13]: Check if it's main thread and throw if needed

**Review by sirambd — CHANGES_REQUESTED** [review]

File: rich-html-editor/src/main/java/com/infomaniak/lib/richhtmleditor/EditorReloader.kt [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+     * override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+     *     super.onViewCreated(view, savedInstanceState)
+     *     lifecycleScope.launch {
+     *         viewModel.editorReloader.load(editor, "<p>Hello World</p>")
+     *     }
+     * }
+     * ```
+     */
+    suspend fun load(editor: RichHtmlEditorWebView, defaultHtml: String) {
+        if (Looper.myLooper() != Looper.getMainLooper()) throw error("The load method needs to be called on the main thread")
```
Comment (sirambd) [inline rich-html-editor/src/main/java/com/infomaniak/lib/richhtmleditor/EditorReloader.kt:46]: The `error` method already uses throw, so there's no need to throw agin

**Review by sirambd — CHANGES_REQUESTED** [review]

**Review by sirambd — APPROVED** [review]

**Review by KevinBoulongne — APPROVED** [review]

**Pull request merged**